internal/group/group.service.go

```diff
@@ -33,58 +37,384 @@ func (s *serviceImpl) FindOne(ctx context.Context, in *proto.FindOneGroupRequest
 	cacheKey := fmt.Sprintf("group:%s", in.UserId)
 	var cachedGroup proto.Group
 
-	// try to retreive from cache
+	// Try to retrieve from cache
 	err := s.cache.GetValue(cacheKey, &cachedGroup)
 	if err == nil {
 		s.log.Info("Group found in cache", zap.String("user_id", in.UserId))
 		return &proto.FindOneGroupResponse{Group: &cachedGroup}, nil
 	}
 
+	// If not found in cache, fetch from database
 	userUUID, err := uuid.Parse(in.UserId)
 	if err != nil {
-		return nil, fmt.Errorf("invalid UUID format: %v", err)
+		s.log.Error("Invalid UUID format", zap.String("user_id", in.UserId), zap.Error(err))
```

Comment:
`s.log.Named("FindOne").Error("Parse: Invalid UUID format ", zap.String("user_id", in.UserId), zap.Error(err))`
let's do something like this instead. It is easier to see which inside method failed

Reply:
please apply to all other logs